lib/policies/redirectPolicy.ts

```diff
@@ -44,23 +44,34 @@ function handleRedirect(
   const locationHeader = response.headers.get("location");
   if (
     locationHeader &&
-    (status === 300 || status === 307 || (status === 303 && request.method === "POST")) &&
-    (!policy.maxRetries || currentRetries < policy.maxRetries)
-  ) {
+    (status === 300 || status === 302 || status === 307 || (status === 303 && request.method === "POST")) &&
+    (request.redirectLimit && currentRetries < request.redirectLimit)) {
```

Comment:
seems like we no longer consider the policy setting for `maxRetries`? While I'm fine having the request's `redirectLimit` take precedence, I don't think we want to regress support for setting a limit on the policy level.

Reply:
I wasn't sure if I should respect maxRetires or remove it:

We're not actually regressing anything, because the call to `node-fetch` always used the deault redirect settings, so node-fetch always followed the redirects before the response got back to this policy. Therefore this policy code **never** actually did anything and `maxReties` never had any effect on the outcome.

The node-fetch redirect behaviour is only overridden if `redirectLimit` is defined, so in other cases the default node-fetch redirect limit of `20` will apply.

Reply:
Sure, I understand that there was a bug before where the policy didn't function as intended, but I don't think it makes sense to continue the buggy behavior in a new way. If we don't want this to be configurable in the policy, I think we should remove it from the signature. This technically is a breaking change, though maybe it doesn't matter as much if it never worked...